### PR TITLE
feat: [SIG-566]: Added message to alert user about their past due - subscription status

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

Added alert message when the subscription status is `past_due`

image

#### Related Issues / PR's

https://github.com/SigNoz/engineering-pod/issues/1177

#### Screenshots

NA

### CODE REVIEW

File: frontend/src/container/BillingContainer/BillingContainer.tsx [hunk 1]
```diff
@@ -186,6 +191,8 @@ export default function BillingContainer(): JSX.Element {
 		[licensesData?.payload?.onTrial],
 	);
 
+	const isSubscriptionPastDue = apiResponse.subscriptionStatus === 'past_due';
```
Comment (Rajat-Dabade) [inline frontend/src/container/BillingContainer/BillingContainer.tsx:194]: NIT: `past_due` to a constant.

**Review by Rajat-Dabade — APPROVED** [review]
One minor comment. Rest LGTM 🚀

**Review by Vikrant2520 — APPROVED** [review]

File: frontend/src/container/BillingContainer/BillingContainer.tsx (hunk 1 above)
Comment (SagarRajput-7) [reply frontend/src/container/BillingContainer/BillingContainer.tsx]: done

**Review by Rajat-Dabade — APPROVED** [review]

**Review by Rajat-Dabade — DISMISSED** [review]

File: frontend/public/locales/en/billings.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+	"subscription_past_due": "Your subscription status is past due",
```
Comment (YounixM) [inline frontend/public/locales/en/billings.json:2]: @SagarRajput-7: The message copy sounds incomplete. We need to provide additional info here.

Comment (YounixM) [reply frontend/public/locales/en/billings.json:2]: `Your subscription status indicates that payment is past due. Please renew your subscription to continue enjoying uninterrupted service.`

**Comment by YounixM** [comment]
LGTM except the message copy.

**Review by YounixM — APPROVED** [review]

File: frontend/public/locales/en/billings.json (hunk 2 above)
Comment (SagarRajput-7) [reply frontend/public/locales/en/billings.json:2]: Done, added suggested changes

**Review by Rajat-Dabade — APPROVED** [review]

File: frontend/public/locales/en/billings.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+	"days_remaining": "days remaining in your billing period.",
+	"billing": "Billing",
+	"manage_billing_and_costs": "Manage your billing information, invoices, and monitor costs.",
```
Comment (Vikrant2520) [inline frontend/public/locales/en/billings.json:4]: can remove the `,` after invoices

File: frontend/src/container/BillingContainer/BillingContainer.tsx [hunk 4]
```diff
@@ -342,14 +355,27 @@ export default function BillingContainer(): JSX.Element {
 		[apiResponse, billAmount, isLoading, isFetchingBillingData],
 	);
 
+	const { Text } = Typography;
+	const subscriptionPastDueMessage = (): JSX.Element => (
+		<Typography>
+			{`We were not able to process payments for your account. Please update your card details `}
```
Comment (Vikrant2520) [inline frontend/src/container/BillingContainer/BillingContainer.tsx:361]: move to strings

**Review by Vikrant2520 — APPROVED** [review]

File: frontend/src/container/BillingContainer/BillingContainer.tsx (hunk 4 above)
Comment (SagarRajput-7) [reply frontend/src/container/BillingContainer/BillingContainer.tsx:361]: will do it in next PR

File: frontend/public/locales/en/billings.json (hunk 3 above)
Comment (SagarRajput-7) [reply frontend/public/locales/en/billings.json:4]: will do it in next PR

**Pull request merged**